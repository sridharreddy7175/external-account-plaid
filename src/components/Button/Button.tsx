import React, { FC, ReactNode, MouseEvent } from "react";

import { Box, Button as ThemeUibutton } from "theme-ui";

export interface ButtonProps {
    onClick?: (e: MouseEvent<ReactNode>) => void;
    buttonName: any;
    buttonStyles: any;
    disabled?: boolean;
    sx?: any;
}

const Button = (props: ButtonProps) => {
    return (
        <Box>
            <ThemeUibutton
                sx={props.sx}
                style={props.buttonStyles}
                onClick={props.onClick}
                disabled={props.disabled}
            >
                {props.buttonName}
            </ThemeUibutton>
        </Box>
    );
};

export default Button;
