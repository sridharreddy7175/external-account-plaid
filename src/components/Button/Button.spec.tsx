import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Button from "./Button";
import renderer from "react-test-renderer";

describe("<Button />", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Button buttonName="History" buttonStyles={{ fontSize: "12px" }} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

// describe("<Button onClick />", () => {
//   it("renders correctly", () => {
//     const tree = renderer.create(<Button onClick={(f: any) => f} />).toJSON();
//     expect(tree).toMatchSnapshot();
//   });
// });
