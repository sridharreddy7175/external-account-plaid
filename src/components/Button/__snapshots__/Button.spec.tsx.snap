// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-vurnku{box-sizing:border-box;margin:0;min-width:0;}",
      }
    }
    data-emotion="css vurnku"
  />,
  <div
    className="css-vurnku"
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-vhv7r1{box-sizing:border-box;margin:0;min-width:0;-webkit-appearance:none;-moz-appearance:none;-ms-appearance:none;appearance:none;display:inline-block;text-align:center;line-height:inherit;-webkit-text-decoration:none;text-decoration:none;font-size:inherit;padding-left:16px;padding-right:16px;padding-top:8px;padding-bottom:8px;color:white;background-color:primary;border:0;border-radius:4px;}",
        }
      }
      data-emotion="css vhv7r1"
    />
    <button
      className="css-vhv7r1"
      style={
        Object {
          "fontSize": "12px",
        }
      }
    >
      History
    </button>
  </div>,
]
`;
